### PR TITLE
[Pytest] Convert CRM test to pytest

### PR DESCRIPTION
**Pull request by yvolynets-mlnx**
### Description of PR

Summary: Converted CRM tests to pytest
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [+] Test case(new/improvement)

#### What is the motivation for this PR?
To Convert ansible CRM test cases to pytest.

Also it fixes the following issues:
**1.** Check that expected error messages are found (negative tests only).
"msg": "Expected error messages are not found while running crm / bash -c \"crm config thresholds ipv4 nexthop type percentage & & crm config thresholds ipv4 nexthop low 2 & & crm config thresholds ipv4 nexthop high 3\""

**2.** If after previous test there are some FDB entries left, CRM can fail if FDB will expire during test

Flow:
CRM test runs
CRM test has strict logic about resources, which can fail under dynamic conditions in testbed, especially when ASIC uses shared resources for NEIGH, ROUTE, etc..

2.1 Before fail:
1636 Jul 14 20:35:55.259092 [DUT] WARNING swss#orchagent: :- checkCrmThresholds: IPV4_NEIGHBOR THRESHOLD_EXCEEDED for TH_USED 1% Used count 504 free count 44000

2.2 Then FDB expires (sairedis.rec):
2019-07-14.20:35:55.748920|n|fdb_event|[{"fdb_entry":"{\"bvid\":\"oid:0x2600000000059d\",\"mac\":\"00:0A:0B:44:05:06\",\"switch_id\":\"oid:0x21000000000000\"}","fdb_event":"SAI_FDB_EVENT_AGED","list":[{"id":"SAI_FDB_ENTRY_ATTR_BRIDGE_PORT_ID","value":"oid:0x3a00000000080c"}, ...........bla bla

2.3 Then crmorch polls resources again
1637 Jul 14 20:35:56.294387 [DUT] WARNING swss#orchagent: :- checkCrmThresholds: IPV4_NEIGHBOR THRESHOLD_EXCEEDED for TH_USED 1% Used count 504 free count 44074

NOTE: 74 entries more.

Than in the syslog during test ("crm config thresholds ipv4 neighbor type free && crm config thresholds ipv4 neighbor low 44000 && crm config thresholds ipv4 neighbor high 44001") CRM does not produce LOG message because counters have changed:

```
Jul 14 20:36:45.383915 [DUT] INFO start-LogAnalyzer-crm.2019-07-14-20:36:33
Jul 14 20:36:45.384682 [DUT] INFO 
Jul 14 20:36:46.080709 [DUT] INFO ansible-<stdin>: Invoked with warn=True executable=None chdir=None _raw_params=bash -c "crm config thresholds ipv4 neighbor type free && crm config thresholds ipv4 neighbor low 44000 && crm config thresholds ipv4 neighbor high 44001" removes=None creates=None _uses_shell=False
Jul 14 20:36:50.178662 [DUT] INFO ansible-<stdin>: Invoked with warn=True executable=None chdir=None _raw_params=sed -i 's/^/#/g' /etc/cron.d/logrotate removes=None creates=None _uses_shell=True
Jul 14 20:36:50.548356 [DUT] INFO ansible-<stdin>: Invoked with warn=True executable=None chdir=None _raw_params=pgrep -f logrotate removes=None creates=None _uses_shell=True
Jul 14 20:36:50.967225 [DUT] INFO ansible-<stdin>: Invoked with directory=/var/log start_string=start-LogAnalyzer-crm.2019-07-14-20:36:33 target_filename=/tmp/syslog file_prefix=syslog
Jul 14 20:36:51.620078 [DUT] INFO ansible-<stdin>: Invoked with warn=True executable=None chdir=None _raw_params=sed -i 's/^#//g' /etc/cron.d/logrotate removes=None creates=None _uses_shell=True
Jul 14 20:36:52.420865 [DUT] INFO ansible-<stdin>: Invoked with warn=True executable=None chdir=/tmp _raw_params=python /tmp/loganalyzer.py --action analyze --logs /tmp/syslog --run_id crm.2019-07-14-20:36:33 --out_dir /tmp/crm.2019-07-14-20:36:33 -m loganalyzer_common_match.txt -i loganalyzer_common_ignore.txt -e expect_crm_th_clear -v removes=None creates=None _uses_shell=True
Jul 14 20:36:52.466446 [DUT] INFO end-LogAnalyzer-crm.2019-07-14-20:36:33
Jul 14 20:36:52.467180 [DUT] INFO 
Jul 14 20:36:53.014169 [DUT] INFO ansible-<stdin>: Invoked with warn=True executable=None chdir=None _raw_params=cat "/tmp/crm.2019-07-14-20:36:33/summary.loganalysis.crm.2019-07-14-20:36:33.log" removes=None creates=None _uses_shell=True
Jul 14 20:36:53.486832 [DUT] INFO ansible-<stdin>: Invoked with warn=True executable=None chdir=None _raw_params=grep "TOTAL EXPECTED" "/tmp/crm.2019-07-14-20:36:33/summary.loganalysis.crm.2019-07-14-20:36:33.log" | sed -n "s/TOTAL EXPECTED MATCHES:[[:space:]]*//p" removes=None creates=None _uses_shell=True
Jul 14 20:36:53.837229 [DUT] INFO ansible-<stdin>: Invoked with warn=True executable=None chdir=None _raw_params=grep "TOTAL MATCHES" "/tmp/crm.2019-07-14-20:36:33/summary.loganalysis.crm.2019-07-14-20:36:33.log" | sed -n "s/TOTAL MATCHES:[[:space:]]*//p" removes=None creates=None _uses_shell=True
Jul 14 20:36:55.419405 [DUT] INFO ansible-<stdin>: Invoked with warn=True executable=None chdir=None _raw_params=bash -c "crm config thresholds ipv4 neighbor type percentage && crm config thresholds ipv4 neighbor low 70 && crm config thresholds ipv4 neighbor high 85" removes=None creates=None _uses_shell=False
Jul 14 20:36:57.255140 [DUT] INFO ansible-<stdin>: Invoked with warn=True executable=None chdir=None _raw_params=crm config polling interval 300 removes=None creates=None _uses_shell=False


```

#### How did you do it?
Converted existed CRM tests which are based on ansible, to pytest based.

#### How did you verify/test it?
Tested on the local setup.

py.test --inventory "../ansible/inventory, ../ansible/veos" --host-pattern [DUT] --module-path ../ansible/library/ --testbed [DUT]-[TOPO] --testbed_file ../ansible/testbed.csv --assert plain --log-cli-level info --show-capture=no --showlocals -v -rs crm/test_crm.py
======================================================================================== 11 passed in 3491.39 seconds =========================================================================================

### CODE REVIEW

**Comment by yxieca** [comment]
@yvolynets-mlnx can you attach a test result to the PR?

File: tests/crm/test_crm.py [hunk 1]
```diff
@@ -0,0 +1,914 @@
...
+                "test_crm_nexthop": [],
+                "test_crm_neighbor": [],
+                "test_crm_nexthop_group": [],
+                "test_acl_entry": [],
+                "test_acl_counter": [],
+                "test_crm_fdb_entry": [],
+                "crm_cli_res": None,
+                "wait": 0}
+
+
```
Comment (yxieca) [inline tests/crm/test_crm.py:43]: Please add topology marker like https://github.com/Azure/sonic-mgmt/blob/master/tests/test_interfaces.py#L5

**Comment by prsunny** [comment]
@volodymyrsamotiy for viz

File: tests/crm/test_crm.py [hunk 2]
```diff
@@ -0,0 +1,914 @@
...
+@pytest.fixture(scope="module", autouse=True)
+def crm_thresholds(duthost):
```
Comment (NazarTkachuk) [inline tests/crm/test_crm.py:45]: Is it any reason why we store fixtures in test suite file?
My suggestion:
if we have local `conftest.py` we should there all fixtures and in test suite file we should store test functions and related helpers functions

File: tests/crm/test_crm.py (hunk 1 above)
Comment (yvolynets-mlnx) [reply tests/crm/test_crm.py:47]: Fixed

File: tests/crm/test_crm.py [hunk 3]
```diff
@@ -0,0 +1,914 @@
...
+    return res
+
+@pytest.fixture(scope="module", autouse=True)
+def crm_interface(duthost, testbed):
+    """ Return tuple of two DUT interfaces """
+    mg_facts = duthost.minigraph_facts(host=duthost.hostname)["ansible_facts"]
+    if testbed["topo"]["name"] == "t1":
+        crm_intf1 = mg_facts["minigraph_interfaces"][0]["attachto"]
+        crm_intf2 = mg_facts["minigraph_interfaces"][2]["attachto"]
+    elif testbed["topo"]["name"] in ["t0", "t1-lag", "t0-52", "t0-56", "t0-64", "t0-116"]:
```
[the remark this answers is not in the document]
Comment (yvolynets-mlnx) [reply tests/crm/test_crm.py]: Fixed

File: tests/crm/test_crm.py (hunk 2 above)
Comment (yvolynets-mlnx) [reply tests/crm/test_crm.py]: In exited test suites there are many files which has defined fixtures inside test suite files.
However I agree with your suggestion, now its fixed.

**Pull request merged**